src/oop/special_class_method/hard/q12/MainTest.java

```diff
@@ -0,0 +1,213 @@
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        _assertEquals(value1, testClass.getFieldValue(classInstance, attributeName1),
+                "Your " + className + " constructor does not initialize the " + attributeName1 + " attribute to the correct value.");
+        _assertEquals(value2, testClass.getFieldValue(classInstance, attributeName2),
+                "Your " + className + " constructor does not initialize the " + attributeName2 + " attribute to the correct value.");
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetYearMadeMethod(int value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName1, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"),
+                "Your " + getAttributeMethodName1 + " method does not have the correct visibility modifier.");
+        _assertEquals(value1, getMethodOutput,
+                "Your " + getAttributeMethodName1 + " method does not return the value of the " + attributeName1 + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetColourMethod(int value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName2, testInstance);
+        assertTrue(testClass.hasModifier(getAttributeMethodName2, null, "public"),
+                "Your " + getAttributeMethodName2 + " method does not have the correct visibility modifier.");
+        _assertEquals(value2, getMethodOutput,
+                "Your " + getAttributeMethodName2 + " method does not return the value of the " + attributeName2 + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("updateIntegerInputProvider")
+    public void correctSetYearMadeMethod(int initialiValue, String value2, int updatedValue) throws Throwable {
+        Object[][] instantiationArguments = {
+                {initialiValue, int.class},
+                {value2, String.class}
+        };
+        Object testInstance = testClass.createInstance(instantiationArguments);
+        Object[][] setMethodArguments = {
+                {updatedValue, int.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                int.class
+        };
+        assertTrue(testClass.hasMethod(setAttributeMethodName1, methodModifierClasses, Void.TYPE),
+                "Your " + setAttributeMethodName1 + " method does not have the correct name, return type, or parameters.");
```

Comment:
split this assert into two so the return value can have it's own assert. Do this in the other set method test as well.